File: Problems/Algorithms/136. Single Number/SingleNumber.java
class Solution {
    public int singleNumber(int[] nums) {
        HashMap<Integer, Integer> count = new HashMap<>();
        
        for (int num : nums) {
            count.put(num, count.getOrDefault(num, 0) + 1);
        }
        
        for (int num : nums) {
            if (count.get(num) == 1) return num;
        }
        
        return -1;
    }
}

File: Problems/Algorithms/136. Single Number/SingleNumber.kt
class Solution {
    fun singleNumber(nums: IntArray): Int {
        val count = mutableMapOf<Int, Int>()
        
        for (num in nums) {
            count.put(num, count.getOrDefault(num, 0) + 1)
        }
        
        for (num in nums) {
            if (count[num] == 1) return num
        }
        
        return -1
    }
}

